installation/src/Console/InstallCommand.php

```diff
@@ -0,0 +1,360 @@
+            $this->ioStyle->error('configuration.php already present! Nothing to install, exiting.');
+            return Command::FAILURE;
+        }
+
+        /* @var CliInstallationApplication $app */
+        $app = $this->getApplication();
+
+        /** @var ChecksModel $checkModel */
+        $checkModel = $app->getMVCFactory()->createModel('Checks', 'Installation');
+        $this->ioStyle->write('Checking system requirements...');
+
+        if (!$checkModel->getPhpOptionsSufficient()) {
+            $options = $checkModel->getPhpOptions();
+
+            foreach ($options as $option) {
+                if (!$option->state) {
+                    $this->ioStyle->error($option->notice);
+
+                    return Command::FAILURE;
+                }
+            }
+        }
+
+        $this->ioStyle->writeln('OK');
+
+        // Collect the configuration
+        $this->ioStyle->write('Collecting configuration...');
+        $cfg                         = $this->getCLIOptions();
+        $cfg['db_pass_plain']        = $cfg['db_pass'];
+        $cfg['admin_password_plain'] = $cfg['admin_password'];
+        $cfg['language']             = 'en-GB';
+        $cfg['helpurl']              = 'https://help.joomla.org/proxy?keyref=Help{major}{minor}:{keyref}&lang={langcode}';
+        $this->ioStyle->writeln('OK');
+
+        /** @var SetupModel $setupModel */
+        $setupModel = $app->getMVCFactory()->createModel('Setup', 'Installation');
+
+        // Validate DB connection
+        $this->ioStyle->write('Validating DB connection...');
+        $setupModel->storeOptions($cfg);
+        $setupModel->validateDbConnection();
+        $this->ioStyle->writeln('OK');
+
+        /** @var DatabaseModel $databaseModel */
+        $databaseModel = $app->getMVCFactory()->createModel('Database', 'Installation');
+
+        // Create and fill database
+        $this->ioStyle->write('Creating and filling database...');
+        $databaseModel->createDatabase();
+        $db = $databaseModel->initialise();
+
+        $files = [
+            'populate1' => 'base',
+            'populate2' => 'supports',
+            'populate3' => 'extensions',
+            'custom1'   => 'localise',
+            'custom2'   => 'custom',
+        ];
+
+        foreach ($files as $step => $schema) {
+            $serverType = $db->getServerType();
+
+            if (\in_array($step, ['custom1', 'custom2']) && !is_file('sql/' . $serverType . '/' . $schema . '.sql')) {
+                continue;
+            }
+
+            $databaseModel->createTables($schema);
+        }
+
+        $this->ioStyle->writeln('OK');
+
+        /** @var \Joomla\CMS\Installation\Model\ConfigurationModel $configurationModel */
+        $configurationModel = $app->getMVCFactory()->createModel('Configuration', 'Installation');
+
+        // Attempt to setup the configuration.
+        $this->ioStyle->write('Writing configuration.php and doing additional setup...');
```

Comment:
not sure what the additional setup is but "doing" is not a good word to use. Also adds the required space for the ellipsis and uses the utf8 character instead of three dots

```suggestion
        $this->ioStyle->write('Writing configuration.php and additional setup …');
```